code/modules/recycling/disposal/bin.dm

```diff
@@ -45,29 +45,26 @@
 
 	return INITIALIZE_HINT_LATELOAD //we need turfs to have air
 
-/obj/machinery/disposal/proc/on_rat_rummage(mob/living/L)
+/obj/machinery/disposal/proc/on_rat_rummage(mob/living/simple_animal/hostile/regalrat/king)
```

Comment:
Signal handlers cannot call methods that sleep, in your case--`do_mob`.

Take out that behavior to a separate proc and call `INVOKE_ASYNC` on that. Searching `INVOKE_ASYNC` in the codebase should give you a good guide.

Reply:
Ok I changed it, but I'm not it's supposed to look like this.

Reply:
Looks right to me, but Rummage_Trash should be on the disposal object, not the rat.

Reply:
I thought I'd put all rat related functions in regalrat, so that if someone wants to change stuff about it, they don't need to search through code for the specific function.